SelfDriveEnv.py

```diff
@@ -268,6 +320,32 @@ def render(self):
         pygame.display.flip()
         self.handle_events()
     
+    def calc_pos(self):
+        for row in self.track:
+            for box in row:
+                if box.start_finish == "start":
+                    self.start_locs.append(box.dimensions[:2])
+                elif box.start_finish == "finish":
+                    self.finish_locs.append(box.dimensions[:2])
+        if len(self.start_locs) > 1:
+            x, y = 0, 0
+            for coord in self.start_locs:
+                x += coord[0]
+                y += coord[1]
+            x /= len(self.start_locs)
+            y /= len(self.start_locs)
+            self.start_locs = [x,y]
+        else: self.start_locs = list(self.start_locs)
+        if len(self.finish_locs) > 1:
+            x, y = 0, 0
+            for coord in self.finish_locs:
+                x += coord[0]
+                y += coord[1]
+            x /= len(self.finish_locs)
+            y /= len(self.finish_locs)
+            self.finish_locs = [x,y]
```

Comment:
This is duplicated logic. If you end up refactoring it as you mentioned above, I think you should extract the duplicated parts into a single function and just call it twice.

Reply:
Okie done